web/src/components/BarcodeChart.vue

```diff
@@ -0,0 +1,203 @@
+<template >
+  <div :id="identifier" class="barcodeChart">
+    <component :is="'style'" type="text/css">
+      .marked {
+      fill: var(--markedbg);
+      }
+      <template v-for="item in hoveringSelections">
+        .marked.{{ item }} {
+        fill: var(--hoveringbg);
+        }
+      </template>
+      <template v-for="item in selectedSelections">
+        .marked.{{ item }} {
+          fill: var(--selectedbg);
+        }
+      </template>
+    </component>
+    <svg class="svg-content-responsive">
+      <rect
+        ref="scrollHighlighter"
+        id="page-scroll-highlighter"
+        :y="scrollOffset"
+      >
+      </rect>
+    </svg>
+  </div>
+</template>
+
+<script lang="ts">
+
+import { Component, Prop, Vue, Watch } from "vue-property-decorator";
+import { Selection } from "@/api/api";
+import { constructID } from "@/util/OccurenceHighlight";
+import * as d3 from "d3";
+
+@Component
+export default class BarcodeChart extends Vue {
+  @Prop() selections!: Array<Selection>;
+  @Prop() sideIdentifier!: string;
+  @Prop() maxLines!: number;
+  @Prop() lines!: number;
+  @Prop() documentScrollFraction!: number;
+  @Prop() amountOfLinesVisible!: number;
+  @Prop() selectedSelections!: Array<string>;
+  @Prop() hoveringSelections!: Array<string>;
+
+  chart: {
+    width: number;
+    height: number;
+  } = {
+    width: 40,
+    height: 1000,
```

Comment:
We don't want to hardcode this. I would let this component resize using CSS and bind the width and height of the chart.

Reply:
The barcode chart actual display height and width are currently controlled by css. The numbers displayed here are the width and the height of the actual svg coordinate system. Controlling these two number via css is not possible nor does it make much sense.

Reply:
If you can choose these values, would it then make sense to use the amount of lines as height and a width of 1?

Reply:
Yes it would, this has now been changed